src/sid/contacts.py

```diff
@@ -433,28 +460,32 @@ def _calculate_infections_by_random_contacts(
                     random_contacts[j, cm] -= 1
 
                     if infectious[i] and not immune[j]:
+                        virus_strain_i = virus_strain[i]
+                        virus_multiplier_i = virus_strains_multipliers[virus_strain_i]
                         is_infection = boolean_choice(
-                            infection_probability_multiplier[j]
+                            infection_probability_multiplier[j] * virus_multiplier_i
```

Comment:
Could use same terminology as I suggested above. Instead of `individual_infection_risk` we could use `adjusted_individual_infection_risk` to make clear that the base probability is already taken care of by the downsampling.